aptos-move/move-examples/token_objects/sources/hero.move

```diff
@@ -0,0 +1,287 @@
+module token_objects::hero {
+    use std::error;
+    use std::option::{Self, Option};
+    use std::signer;
+    use std::string::{Self, String};
+
+    use aptos_framework::object::{Self, CreatorRef, ObjectId};
+
+    use token_objects::collection;
+    use token_objects::token;
+
+    const ENOT_A_HERO: u64 = 1;
+    const ENOT_A_WEAPON: u64 = 2;
+    const ENOT_A_GEM: u64 = 3;
+    const ENOT_CREATOR: u64 = 4;
+    const EINVALID_WEAPON_UNEQUIP: u64 = 5;
+    const EINVALID_GEM_UNEQUIP: u64 = 6;
+
+    struct OnChainConfig has key {
+        collection: String,
+        mutability_config: token::MutabilityConfig,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct Hero has key {
+        armor: Option<ObjectId>,
+        gender: String,
+        race: String,
+        shield: Option<ObjectId>,
+        weapon: Option<ObjectId>,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct Armor has key {
+        defense: u64,
+        gem: Option<ObjectId>,
+        weight: u64,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct Gem has key {
+        attack_modifier: u64,
+        defense_modifier: u64,
+        magic_attribute: String,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
```

Comment:
I mid term we should add some syntactic sugar to avoid repetition of `#[resource_group_member(group = aptos_framework::object::ObjectGroup)]`.

E.g. 

`struct Shield has object {}` or something like that

Reply:
That's a deep compiler change, but I don't disagree... the problem is that syntax can be confusing if I want it to be `key` but not in the object group

Reply:
If you want it to be key but not in the object group you can simply add `has key`.
If it's easier we can also do something like `#[object]` which is still a lot less verbose than always typing out `#[resource_group_member(group = aptos_framework::object::ObjectGroup)]` for every resource in the object model.

I'm just thinking of a future where the object model is so popular that 80% of structs in a given dApp are object-native structs and having to repeat this quite long annotation seems ugly and noisy.

Reply:
I hear you, we should definitely iterate in this space. I don't want to rush to a quick fix if we can actually think through more considerations and land on a great solution.

Reply:
yeah, agree. I think since adding this is purely syntactic sugar it probably can be done a bit later.